src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,125 @@
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.Optional;
+import mate.academy.model.Book;
+import mate.academy.util.ConnectionUtil;
+
+@Dao
+public class BookDaoImpl implements BookDao {
+
+    @Override
+    public Book save(Book book) {
+        String sql = "INSERT INTO book (title, year) VALUES (?, ?)";
+        try (Connection connection = getConnection(); PreparedStatement statement
+                = connection.prepareStatement(sql, Statement.RETURN_GENERATED_KEYS)) {
+            statement.setString(1, book.getTitle());
+            statement.setInt(2, book.getPrice());
+
+            int affectedRows = statement.executeUpdate();
+            if (affectedRows < 1) {
+                throw new RuntimeException("Expected to insert at leas one row, "
+                        + "but inserted 0 rows.");
+            }
+
+            ResultSet generatedKeys = statement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                Long id = generatedKeys.getObject(1, Long.class);
+                book.setId(id);
+            }
+        } catch (SQLException e) {
+            throw new RuntimeException("Can not add new book: " + book, e);
+        }
+        return book;
+    }
+
+    @Override
+    public Book get(Long id) {
+        String sql = "SELECT * FROM book WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection(); PreparedStatement statement
+                = connection.prepareStatement(sql)) {
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                String title = resultSet.getString("title");
+                Integer price = resultSet.getObject("price", Integer.class);
+                Book book = new Book();
+                book.setId(id);
+                book.setTitle(title);
+                book.setPrice(price);
+                return book;
+            }
+
+        } catch (SQLException e) {
+            throw new RuntimeException("Can not create a connection to the DB", e);
+        }
+        return null;
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        String sql = "SELECT * FROM books WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection(); PreparedStatement statement
+                = connection.prepareStatement(sql)) {
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                return Optional.of(findBook(resultSet));
+
+            }
+        } catch (SQLException e) {
+            throw new RuntimeException("Can not find a book by id: " + id, e);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public Book update(Book book) {
+        String sql = "INSERT INTO books (title, price) VALUES (?, ?)";
```

Comment:
The SQL statement should be an UPDATE query, not INSERT, when updating a book. Use 'UPDATE books SET title = ?, price = ? WHERE id = ?'.